### ISSUE TITLE
False positive error when `type(None)` or `NoneType` is used as second argument to `issubclass`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#6186]
Pyright shouldn't generate an error when `type(None)` or `NoneType` is used in the second argument to `issubclass`.

```python
issubclass(int, type(None))
```

Furthermore, this should work for type narrowing, but it doesn't currently.

```python
def func(x: type[None] | type[int]):
    if issubclass(x, type(None)):
        reveal_type(x, expected_text="type[None]")
    else:
        reveal_type(x, expected_text="type[int]")
```

This is related to a bug filed in the mypy issue tracker: https://github.com/python/mypy/issues/16279

**Comment by erictraut** [comment]
This is included in pyright 1.1.332, which I just published. It will also be included in a future release of pylance.
